src/python_testing/tc_mode_base.py

```diff
@@ -0,0 +1,112 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from mobly import asserts
+from chip.clusters.Types import NullValue
+
+logger = logging.getLogger(__name__)
+
+
+class TC_MODE_BASE:
+
+    def initialize_tc_base(self, endpoint, attributes, requested_cluster, cluster_objects, *args):
+        self.commonTags = {0x0: 'Auto',
```

Comment:
where do these magic constants come from? Do we have them from codegen or could we reference spec locations?

Reply:
They are defined in the spec for most of the "mode" clusters and seem to be common between them, this seems to be the spec (https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/modebase_common.adoc). However it seems that those values are part of the enum of each cluster so it is actually not needed to add them in a different structure.